db/version_set.cc

```diff
@@ -3825,13 +3838,17 @@ Status VersionSet::ProcessManifestWrites(
         } else if (group_start != std::numeric_limits<size_t>::max()) {
           group_start = std::numeric_limits<size_t>::max();
         }
-        Status s = LogAndApplyHelper(last_writer->cfd, builder, e, mu);
-        if (!s.ok()) {
-          // free up the allocated memory
-          for (auto v : versions) {
-            delete v;
+        if (e->IsWalManipulation()) {
+          LogAndApplyWalHelper(e);
```

Comment:
If you let `LogAndApplyWalHelper` return `Status`, then the error handling logic can be shared, saving some indentation.

Reply:
I realized that we can get rid of `LogAndApplyWalHelper`!